torchao/dtypes/uintx/csr_layout.py

```diff
@@ -0,0 +1,266 @@
+    w_csr = weight_tensor.tensor_impl.int_data  # 2-D sparse CSR INT8
+    w_scale = weight_tensor.tensor_impl.scale  # shape: (1,) or (row,)
+
+    # Reshape activations to 2-D (batch*seqlen, in_features)
+    x2d = x_vals_int8.reshape(-1, x_vals_int8.shape[-1]).to(torch.int32)
+
+    # As for now we don't have any kernel for spmm form int8 computation for ARM so we are upscaling it to fp32 for computation
+    # Once we intigrate the kernel in torch then we can use that kernel instead of torch.mm()
+    y_int32 = torch.mm(
+        w_csr.to(torch.float32), x2d.t().to(torch.float32).contiguous()
+    ).t()
+
+    # Dequantise: y = (x_scale * w_scale) * y_int32
+    scale_mat = x_scale.view(-1, 1) * w_scale.view(1, -1)
+
+    y_fp32 = (y_int32.to(torch.float32) * scale_mat).reshape(
+        *x_vals_int8.shape[:-1],  # (B,)
+        y_int32.shape[-1],  # O
+    )
+
+    if bias is not None:
+        y_fp32 += bias
+
+    # Cast back to activation dtype (usually fp32 or bf16)
+    return y_fp32.to(input_tensor.dtype).contiguous()
+
+
+@dataclass(frozen=True)
+class CSRLayout(Layout):
+    """Layout marker for *Compressed Sparse Row* INT8 weights.
+
+    Parameters
+    ----------
+    target_sparsity : float ∈ (0, 1) or *None*
+        If given, this overrides the global env-var and hard-coded default.
+    """
+
+    target_sparsity: Optional[float] = None
+
+    def __post_init__(self):
+        if self.target_sparsity is not None:
+            if not (0.0 < self.target_sparsity < 1.0):
+                raise ValueError(
+                    "CSRLayout(target_sparsity=…) expects a value in (0, 1) "
+                    f"but got {self.target_sparsity}"
+                )
+
+    def pre_process(self, input: torch.Tensor) -> torch.Tensor:
+        """Magnitude‑based pruning prior to CSR packing.
+
+        Parameters
+        ----------
+        input : torch.Tensor
+            A **dense** weight matrix (any dtype) that we intend to quantise
+            and pack as CSR.  The function returns a *copy* with elements below
+            a global magnitude threshold set to zero so that subsequent
+            `torch._to_csr` compression produces the desired sparsity.
+
+        Heuristic
+        ---------
+        *Prune* the smallest‑magnitude values until we reach the *target
+        sparsity* fraction.  The target defaults to **90 %** but can be
+        overridden by setting the argument variable target_sparsity
+        ``TORCHAO_CSR_TARGET_SPARSITY`` (float in 0 – 1).  If the target is
+        outside that range the function becomes a no‑op and simply returns the
+        input unchanged.
+        """
+        import os
+
+        # 1. explicit per-instance override wins
+        if self.target_sparsity is not None:
+            target = self.target_sparsity
+        else:
+            # 2. fall back to env var or 0.9 default
+            target = float(os.getenv("TORCHAO_CSR_TARGET_SPARSITY", "0.9"))
```

Comment:
Do we need an ENV variable for this? I think maybe it would be better to just pass this in the Config.